boards/wch/ch32v003evt/doc/index.rst

```diff
@@ -0,0 +1,115 @@
+
+* CLOCK
+* FLASH
+* :abbr:`GPIO (General Purpose Input Output)`
+* :abbr:`NVIC (Nested Vectored Interrupt Controller)`
+* :abbr:`PWM (Pulse Width Modulation)`
+
+.. figure:: img/wch_ch32v003evt_front.jpg
+     :align: center
+     :alt: WCH CH32V003EVT
+
+     WCH CH32V003EVT (Credit: WCH)
+
+The board is equipped with two LEDs. The `WCH webpage on CH32V003`_ contains
+the processor's information and the datasheet.
+
+Hardware
+********
+
+The QingKe 32-bit RISC-V2A processor of the WCH CH32V003EVT is clocked by an
+external crystal and runs at 48 MHz.
+
+Supported Features
+==================
+
+The wch_ch32v003evt board configuration supports the following hardware
+features:
+
++-----------+------------+----------------------+
+| Interface | Controller | Driver/Component     |
++===========+============+======================+
+| CLOCK     | on-chip    | clock_control        |
++-----------+------------+----------------------+
+| FLASH     | on-chip    | flash                |
++-----------+------------+----------------------+
+| GPIO      | on-chip    | gpio                 |
++-----------+------------+----------------------+
+| PWM       | on-chip    | pwm                  |
++-----------+------------+----------------------+
+
+Other hardware features have not been enabled yet for this board.
+
+Connections and IOs
+===================
+
+LED
+---
+
+* LED = Unconnected. Connect to an I/O pin as needed.
+
+Programming and Debugging
+*************************
+
+Applications for the ``wch_ch32v003evt`` board configuration can be
+built and flashed in the usual way (see :ref:`build_an_application`
+and :ref:`application_run` for more details); however, an external programmer
+is required since the board does not have any on-board debug IC.
+
+The following pins of the external programmer must be connected to the
+following pins on the PCB (see image):
+
+* VCC = VCC (do not power the board from the USB port at the same time)
+* GND = GND
+* SWIO = PD1
+
+Flashing
+========
+
+You can use `minichlink` or `picorvd` to flash the board. Once `minichlink` or
+`picorvd` have been set up, build and flash applications as usual (see
+:ref:`build_an_application` and :ref:`application_run` for more details).
+
+Here is an example for the :zephyr:code-sample:`blinky` application.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/basic/blinky
+   :board: wch_ch32v003evt
+   :goals: build flash
+
+Debugging
+=========
+
+This board can be debugged via OpenOCD or `minichlink` or `picorvd`.
+
+Testing the LED on the WCH CH32V003EVT
+**************************************
+
+There is 1 sample program that allow you to test that the LED on the board is
+working properly with Zephyr:
+
+.. code-block:: console
+
+   samples/basic/blinky
+
+You can build and flash the examples to make sure Zephyr is running
+correctly on your board. The button and LED definitions can be found
+in :zephyr_file:`boards/wch/ch32v003evt/wch_ch32v003evt.dts`.
```

Comment:
If the LED is not connected to anything and needs a jumper, this I would assume this would actually not work, so there should be instructions here for how to get this to work, both in what configuration is needed for the sample/dts and what is needed for the actual hardware. Above states:

> * LED = Unconnected. Connect to an I/O pin as needed.

Reply:
Hmm, there's two variants of the ch32v003evt: this one, which needs a jumper, and [this one](https://www.aliexpress.us/item/1005004964355080.html)

The second one is the one I have, and it has a LED on PD4.

@cnlohr do you happen to know which one is the  ch32v003evt? The second one has "QSZNTEC" written on the back, which is also written on my WCHLink.

Reply:
@nzmichaelh - [that one](https://www.aliexpress.us/item/1005004964355080.html) is the variant that I have as well. 

@kholia - this is looking really great - I'll keep an eye on it 👍

Reply:
Modified to say `LED1 = Unconnected. Connect to an I/O pin (PD4)`. This holds for the official WCH CH32V003EVT board (with square corners and NOT the other board with round corners).